### DISCUSSION TITLE
Serialisation vs Transformation

### DISCUSSION THREAD

**Discussion by sondreso** [equinor/ert#2423] [General]
Today we have both the concepts of serialisation and transformation, and there are a few questions we should discuss (and document) related to this. Should we merge these into one concept, as one could argue that serialisation is a subset of transformation? Are we doing serialisation of numerical/structured data, and transformations when we are handling binary data? Should these be different implementations, but share the same interface? 

Currently we use the mime type to express how we want numerical data to appear on disk, however, we lack this expressiveness for the binary data (file vs directory/unzip etc.).

**Comment by jondequinor** [comment]
Note that mime types can be qualified with a suffix using `+`. E.g. `application/octet-stream+grid`. More https://datatracker.ietf.org/doc/html/draft-ietf-appsawg-media-type-regs-14#section-6
